vocode/streaming/agent/base_agent.py

```diff
@@ -240,14 +260,52 @@ async def process(self, item: InterruptibleEvent[AgentInput]):
                 self.produce_interruptible_event_nonblocking(AgentResponseFillerAudio())
             self.logger.debug("Responding to transcription")
             should_stop = False
+            function_call = None
             if self.agent_config.generate_responses:
-                should_stop = await self.handle_generate_response(
+                should_stop, function_call = await self.handle_generate_response(
```

Comment:
can pass the uuid here; also could do it with the conversation state manager (think about what's best, i don't want to be passing excessive state around)